### PR TITLE
Making normal projection stabler.

### PR DESCRIPTION
**Pull request by emailweixu**
The previous ppo_icubwalk uses state independent standard deviation for action distribution and reaches reward 700 after 10M steps. Using state dependent std can reach higher reward but the training is very unstable. This PR tries to make the training more stable by using a different way of parameterizing Gaussian. In the figure blue and red curves uses new method, orange curve uses old method.
icubwalk-stable-normal-projection

### CODE REVIEW

File: alf/networks/stable_normal_projection_network.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            c -= 1 / (max_std - min_std)
+
+        if inverse_std_transform == 'exp':
+            std_transform = tf.math.exp
+            std_bias_initializer_value = math.log(c)
+        elif inverse_std_transform == 'softplus':
+            std_transform = tf.nn.softplus
+            std_bias_initializer_value = math.log(math.exp(c) - 1)
+        else:
+            raise TypeError("Unsupported std_transform %s" % std_transform)
```
Comment (witwolf) [inline alf/networks/stable_normal_projection_network.py]: raise TypeError("Unsupported std_transform %s" % inverse_std_transform)

File: alf/utils/math_ops.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import gin
+import tensorflow as tf
+"""Various math ops."""
+
+
+@gin.configurable
+def clipped_exp(value, clip_value_min=-20, clip_value_max=2):
```
Comment (witwolf) [inline alf/utils/math_ops.py]: same implementation with `alf.utils.common.std_clip_transform` , should remove one of them

File: alf/networks/stable_normal_projection_network.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        """
+        self._min_std = min_std
+        self._max_std = max_std
+
+        c = 1 / (init_std - min_std)
+        if max_std is not None:
+            c -= 1 / (max_std - min_std)
+
+        if inverse_std_transform == 'exp':
+            std_transform = tf.math.exp
```
Comment (witwolf) [inline alf/networks/stable_normal_projection_network.py]: numeric overflow  is very likely to happen if `tf.math.exp` was apply without clipping

**Comment by Jialn** [comment]
The final reward seems promising, nearly 200 higher than SAC.
